core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -765,9 +765,15 @@ private void loadMetadataForPath(LockedInodePath inodePath)
         && !inodePath.getUri().equals(mRootScheme.getPath())) {
       descendantType = DescendantType.NONE;
     }
+
+    FileSystemMasterCommonPOptions option = NO_TTL_OPTION;
```

Comment:
@HelloHorizon Our plan here at this moment is to change the default behavior to sync-ed files having the correct TTL actions from client option/path conf. Do you have concerns about this? I rmb @tcrain agrees having inherited TTL for sync-ed inodes is a more intuitive default behavior.

Also Baolong found that the current logic is intentional (https://github.com/Alluxio/alluxio/pull/11529). @HelloHorizon Could you help us recall what the intention was? Thx :)

Reply:
Also @ZacBlanco Do you happen to remember what #11529 was for? Thanks!

Reply:
I vaguely recall the issue being that on new inodes being synced that the TTL would be applied in cases where most users would *not* want/expect the TTL to be applied. Here's a scenario:

- HDFS file path: `/a/b/c`
- Alluxio path: `/mnt/hdfs/a/b/c`
- metadata sync on `/mnt/hdfs/a`

Without the referenced PR applied, if the sync was initiated and the path `/mnt/hdfs/a/b/c` was created in Alluxio by the metadata sync and the default TTL option was a `DELETE`, then the file would be created in Alluxio, but then deleted in Alluxio *AND* HDFS after the TTL expires. IIRC in discussion with Gene, this was an undesirable behavior because the file wasn't originally created in Alluxio; it was created in HDFS, so applying the Alluxio TTL could cause unintentional data loss.

Reply:
@ZacBlanco Got it, thanks for the explanation! However I thought the sync-ed file would have by default no TTL (essentially TTL=-1 means not checking). So although the default action is DELETE, the TTL never expires and should never be removed unexpectedly.

Anyways my takeaway from your memory is that as long as files created by sync are not incorrectly affected by the default DELETE behavior, that should be fine. So in this change we will make sure the TTL duration are set correctly reflecting the real intention of the user, namely:
1. -1 by default and no expiry
2. Users can specify from the command by `alluxio.user.file.create.ttl=xx duration`
3. The file can inherit from path conf

Thanks again for your timely response, have a good one :)